## STAT 207 Lab 4: Random Variables - [25 total points] - Solutions

Due: Wednesday, March 10, 11:59pm CST



## Imports

In [1]:
import pandas as pd
import numpy as np

## <u>Question 1</u>: Bernoulli Random Variables

In the previous year the acceptance rate at UIUC was 62\%. Suppose we conduct an experiment where we randomly select one of the UIUC applicants from the previous year and observe whether they were accepted into UIUC. We're interested to know if the applicant was accepted.


The Bernoulli random variable object in Python operates has similar functions to the other random variables that we have discussed in this class. The documentation for the functions that correspond to the bernoulli random variable are shown here.
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bernoulli.html

### 1.1 [2 pts] Define a Bernoulli random variable that models this experiment.
* Make sure you define any parameters that pertain to this random variable as well as define the success level.
(Just put it into words).

x ~ Bern(p = .62)

### 1.2 [2 pts]  Calculate the mean, variance, and standard deviation of this random variable.

$\mu$ = .62       
$\sigma$ = .4853864439804639     
$\sigma ^ 2$ = .2356 


In [2]:
from scipy.stats import bernoulli

In [3]:
mean, variance = bernoulli.stats(p = .62, loc=0, moments= 'mv')

In [4]:
mean
#.62

array(0.62)

In [5]:
variance
#.62(.38)

array(0.2356)

In [6]:
std = variance ** (1/2)
std
#sqrt(variance)

0.4853864439804639

### 1.3 [2 pts]  Generate 1000 random values for this random variable and put them into a pandas dataframe.

In [7]:
sampling_distribution = bernoulli.rvs(p =.62, loc=0, size=1000)
bern_1000 =  pd.DataFrame({'Random Value' : sampling_distribution})


In [8]:
bern_1000

Random Value
0               1
1               1
2               1
3               0
4               1
..            ...
995             1
996             1
997             1
998             1
999             1

[1000 rows x 1 columns]

### 1.4 [2 pts]  Calculate the mean, variance, and standard deviation of these randomly generated random variable values. 

In [9]:
bern_mean = bern_1000.mean()
bern_mean

Random Value    0.622
dtype: float64

In [10]:
bern_var = bern_1000.var()
bern_var

Random Value    0.235351
dtype: float64

In [11]:
bern_std = bern_1000.std()
bern_std

Random Value    0.48513
dtype: float64

### 1.5 [2 pts]  How close are your mean, variance, and standard deviation from 1.4 to the mean, variance, and standard deviation that you calculated in 1.2?

In [12]:
mean_dif = mean - bern_mean
mean_dif

Random Value   -0.002
dtype: float64

In [13]:
var_dif = variance - bern_var
var_dif

Random Value    0.000249
dtype: float64

In [14]:
std_dif = std - bern_std
std_dif


Random Value    0.000256
dtype: float64

They are all very close in values with respective differences of -0.03, 0.007872, 0.008178

## <u>Question 2</u>: More Random Variables

Suppose a gambler has a strategy to keep playing a slot matchine until he wins a round. After he **wins** a round, he stops playing for the day. The probability of **losing** any given slot machine round is 0.99.

### 2.1 [2 pts]  On average, how many rounds of the slot machine will the gambler play before stopping for the day? 



100

In [16]:
from scipy.stats import geom

In [17]:
geom.mean(p = .01)

100.0

### 2.2 [2 pts]  Suppose that it costs $\$1$ to play each round of the slot machine. The gambler always brings $\$50$ with him to play the slot machine each day. If he runs out of this $\$50$ (ie. he plays the game more than 50 times), then he goes to borrow money from his friend and keeps playing the slot machine until he wins. What percent of days does he need to borrow money from this friend?

In [18]:
#p(x >= 51)

1 - geom.cdf(50, p =.01)

0.6050060671375366

60.50% of time he needs to borrow money from his friend

## <u>Question 3</u>: Normal Random Variables

Suppose the distribution of SAT scores for the seniors at a local high school (just math and verbal) has an average of 1000, a standard deviation of 100, and follows a normal distribution. 

### 3.1 [2 pts]  What is the probability that randomly selected senior from the high school scored more than 1250?

0.006209665325776159

In [19]:
from scipy.stats import norm
1 - norm.cdf(1250, loc = 1000, scale = 100)

0.006209665325776159

### 3.2 [2 pts]  What percent of seniors at the school made score between a 900 and 950?

0.1498822847945298

In [20]:
norm.cdf(950, loc = 1000, scale = 100) - norm.cdf(900, loc = 1000, scale = 100)

0.1498822847945298

### 3.3 [3 pts]  Suppose that all seniors at the school who score higher than a 1200 get a scholarship from the county. Suppose that we randomly select a student that we KNOW got the scholarship. GIVEN that we know that the student has the scholarship, what is the probability that this student's SAT score was less than 1300?

#### Hint: You might need to consider some other probability rules that we've encountered in this class.

0.9406641669285728

In [21]:
#p(a | b) = p(a & b)/p(b)

#p(a) = probability that SAT score was less then 1300
p_a = norm.cdf(1300,loc = 1000, scale = 100)

#p(b) = probability that SAT score is higher than 1200
p_b = 1 - norm.cdf(1200,loc = 1000, scale = 100)

#p(a & B) = more than 1200 but less than 1300
p_both = norm.cdf(1300, loc = 1000, scale = 100) - norm.cdf(1200, loc = 1000, scale = 100)


print(p_a)
print(p_b)
print(p_both)

0.9986501019683699
0.02275013194817921
0.021400233916549105


In [22]:
p_both/p_b

0.9406641669285728

## <u>Question 4</u>: Exploring Properties of the Sample Mean Random Variable

Let's revisit our distribution of SAT scores for the seniors at a local high school (just math and verbal) that had an average of 1000, a standard deviation of 100, and followed a normal distribution. 

Let's define **$X_1$=SAT score of a randomly selected high school senior**

### 4.1 [2 pts] What is $E[X_1]$,  $V[X_1]$, and $SD[X_1]$? (looking for actual numbers).

In [23]:
#mean = 1000
#var = 100 ** 2 = 10000
#std = 100

#expected values are the same or approximately the same as population

### <u>Tutorial</u>: Functions of Random Variables

Remember from Unit 7, that we can create new random variables by adding and multiplying random variable by other random variables or other numbers.

For instance, if we have the two following random variables,

**$X_1$=SAT score of a randomly selected high school senior**

**$X_2$=SAT score of a randomly selected high school senior**

then $\bar{X}=\frac{X_1+X_2}{2}$ is also a random variable.

### <u>Tutorial</u>: Properties of the $E[ ]$ function and the $V[]$ function.

* Recall that if we are dealing with a random variable Y that is associated with an experiment, then 
    - if we were to repeat this experiment many, many times, calculate the value of Y for the outcome of each of these experiments, and then take the **average** of these Y values, we would expect this average to be very close to $E[Y]$.
    - if we were to repeat this experiment many, many times, calculate the value of Y for the outcome of each of these experiments, and then take the **variance** of these Y values, we would expect this average to be very close to $V[Y]$.
    

We can think of these $E[ ]$ and $V[]$ as **functions** that also happen to have certain properties.


#### <u>Property of the $E[ ]$ function</u>
For instance, if X and Y ar|e random variables and a and b are coefficients, then
$E[aX+bY] = a\cdot E[X]+b\cdot E[Y]$.

For instance:

$E[7X_1+4X_2] = 7\cdot E[X_1]+4\cdot E[X_2]$.

### 4.2  [2 pts] Calculating $E[\bar{X}]$.
Let's define the following four random variables:

* **$X_1$=SAT score of a randomly selected high school senior**

* **$X_2$=SAT score of a randomly selected high school senior**

* **$X_3$=SAT score of a randomly selected high school senior**

* **$X_4$=SAT score of a randomly selected high school senior**.

Specifically, we say that $X_i \sim N(\mu = 1000, \sigma=100)$ for i=1, i=2, i=3, and i=4.

Finally, let's define one more random variable as:
* **$\bar{X}=\frac{X_1+X_2+X_3+X_4}{4}$**


#### Use the properties from the tutorials above to calculate $E[\bar{X}]$.

This is a proof, so show your work!

$E[\bar{X}]$ = $E[(X_1+X_2+X_3+X_4) / 4]$

= $E[(1/4X_1+1/4X_2+1/4X_3+1/4X_4)]$

= $1/4\cdot E[X_1] + 1/4\cdot E[X_2] + 1/4\cdot E[X_3] + 1/4\cdot E[X_4]$

= $1/4\cdot \mu + 1/4\cdot \mu + 1/4\cdot \mu + 1/4\cdot \mu $

$E[\bar{X}]$ = $\mu$



